File: Hydrogene/Nougat/UIKit/IBAction/IBAction/RootViewController.cs
namespace CSharpiOSIBActionDemo
{
	using UIKit;

	[IBObject]
  public class RootViewController : UIViewController
	{
	
	
	  private UILabel fLabel;

    [IBOutlet]
    public UILabel label
    {
        get { 
           return fLabel;
        }

        set {
            fLabel = value;
        }
		}

		public override id init()
		{
			this = base.initWithNibName("RootViewController") bundle(null);
			if (this != null)
			{
				title = "iOS IBAction Demo";
			}
			return this;
		}

    [IBAction]
    public void buttonTapped(id sender)
    {
       label.text = "Button Was Tapped";
    }
	
	}
}

File: Oxygene/Toffee/UIKit/IBAction/IBAction/RootViewController.pas
namespace OxygeneiOSIBActionDemo;

interface

uses
  UIKit;

type
  [IBObject]
  RootViewController = public class(UIViewController)
  private
  public
    method init: id; override;
    [IBOutlet] label : UILabel;
    [IBAction] method buttonTapped(sender: id);
  end;

implementation

method RootViewController.init: id;
begin
  self := inherited initWithNibName('RootViewController') bundle(nil);
  if assigned(self) then begin
    title := 'iOS IBAction Demo';
  end;
  result := self;
end;

method RootViewController.buttonTapped(sender: id);
begin
  label.text := "Button Was Tapped";
end;

end.
